auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/LdapSecurityRealm.java

```diff
@@ -700,7 +700,9 @@ private LdapIdentity getIdentity(DirContext dirContext, Collection<String> retur
         private String extractRdn(AttributeMapping mapping, final String dn) {
             String valueRdn = mapping.getRdn();
             try {
-                for (Rdn rdn : new LdapName(dn).getRdns()) {
+                LdapName dnName = new LdapName(dn);
+                for (int i = dnName.size() - 1; i >= 0; i--) {
```

Comment:
@rmartinc Just a small suggestion. I think it would be good to add a comment here to explain why we need to use the reverse order.

Reply:
Done! Added a little comment saying that we need reverse (left to right) order.